lib/pharos/reboot_command.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Pharos
+  class RebootCommand < Pharos::Command
+    options :filtered_hosts, :yes?
+
+    def execute
+      Dir.chdir(config_yaml.dirname) do
+        filtered_hosts.size == load_config.hosts.size ? reboot_all : reboot_hosts
+      end
+    end
+
+    def reboot_all
+      confirm_yes!(pastel.bright_yellow("==> Do you really want to reboot all of the hosts in the cluster?"), default: false)
+      reboot_hosts
+    end
+
+    def reboot_hosts
+      start_time = Time.now
+
+      master_hosts = filtered_hosts.select(&:master?).reject(&:local?)
+      worker_hosts = filtered_hosts.select(&:worker?).reject(&:local?)
+      local_hosts  = filtered_hosts.select(&:local?)
+
+      puts pastel.green("==> Sharpening tools ...")
+      cluster_manager.gather_facts
+
+      unless local_hosts.empty?
+        puts "  " + pastel.red("!" * 76)
+        puts pastel.red("    The host will remain cordoned (workloads will not be scheduled on it) after the reboot")
+        puts pastel.red("    To uncordon, you must use: ") + pastel.cyan("pharos exec -c #{config_yaml.filename} -r master -f -- kubectl uncordon #{local_hosts.first}")
```

Comment:
Hmm, it could be possible to schedule uncordoning by using something like:

```console
echo "kubectl uncordon #{localhost.hostname}" | at now + 10 minutes
```

before issuing the reboot command on the localhost